pgxn/hnsw/hnswalg.cpp

```diff
@@ -223,6 +223,7 @@ dist_t fstdistfunc_scalar(const coord_t *x, const coord_t *y, size_t n)
 {
     dist_t 	distance = 0.0;
 
+    #pragma clang loop vectorize(enable)
```

Comment:
Seems silly that we called it "scalar" but then vectorize it :) 

But it's probably fine. Btw, have you benchmarked how this loop vectorize enable thingy works on x86 with AVX2? It might be possible to remove the hand-written intrinsics (I think L2 distance is a case where compiler should be quite good at vectorizing it).